geolocator/example/lib/main.dart

```diff
@@ -154,7 +155,14 @@ class _GeolocatorWidgetState extends State<GeolocatorWidget> {
                             _positionStreamSubscription!.isPaused)
                         ? const Icon(Icons.play_arrow)
                         : const Icon(Icons.pause),
-                    onPressed: _toggleListening,
+                    onPressed: () {
+                      if (positionStreamStarted) {
+                        positionStreamStarted = false;
+                      } else {
+                        positionStreamStarted = true;
+                      }
```

Comment:
You can shorten this a bit by using:
```suggestion
                      positionStreamStarted = !positionStreamStarted;
```

Reply:
Thanks, I thought it was 'cleaner' to make use of a if statement in this case.